doc/conf.py

```diff
@@ -403,6 +416,68 @@ def __call__(self, filename):
         return -version_float
 
 
+def notebook_modification_function(notebook_content, notebook_filename):
+    notebook_content_str = str(notebook_content)
+    warning_template = "\n".join(
+        [
+            "<div class='alert alert-{message_class}'>",
+            "",
+            "# JupyterLite warning",
+            "",
+            "{message}",
+            "</div>",
+        ]
+    )
+
+    message_class = "warning"
+    message = (
+        "Running the scikit-learn examples in JupyterLite is experimental "
+        "and you may encounter some unexpected behavior.\n\n"
+        "The main difference is that imports will take a lot longer than usual, "
+        "for example the first `import sklearn` can take roughly 20-30s.\n\n"
+        "If you notice problems, feel free to open an "
+        "[issue](https://github.com/scikit-learn/scikit-learn/issues/new/choose) about it."
+    )
+
+    markdown = warning_template.format(message_class=message_class, message=message)
+
+    dummy_notebook_content = {"cells": []}
+    add_markdown_cell(dummy_notebook_content, markdown)
+
+    code_lines = []
+
+    if "seaborn" in notebook_content_str:
+        code_lines.append("%pip install seaborn")
+    if "plotly.express" in notebook_content_str:
+        code_lines.extend(["%pip install plotly", "import pandas"])
+    if "skimage" in notebook_content_str:
+        code_lines.append("%pip install scikit-image")
+    if "fetch_" in notebook_content_str:
+        code_lines.extend(
+            [
+                "%pip install pyodide-http",
+                "import pyodide_http",
+                "pyodide_http.patch_all()",
+            ]
+        )
+    if (
+        # TODO(1.4) the default fetch_openml parser will be pandas so any
+        # notebook using fetch_openml will need 'import pandas'
+        'parser="pandas"' in notebook_content_str
+        or "as_frame=True" in notebook_content_str
+    ):
+        code_lines.extend("import pandas")
```

Comment:
Does `JupyterLite` require an "import pandas" before a library can use it?

In any case, I am okay with running "import pandas" all the time to prevent any pandas related issues.

Reply:
My understanding is that there is a bit of magic in Pyodide that if it you type `import matplotlib` in the Pyodide console (or in JupyterLite with a Pyodide kernel in our case) it will "just work". This is the case for all packages that are included inside Pyodide, including pandas.

On the other hand if you are calling a function that does an import inside it it will not work. Examples where this happens:
- `from sklearn.utils import check_matplotlib_support; check_matplotlib_support('test')`
- `%matplotlib inline`
 
The kind of error you get:
```
ModuleNotFoundError: The module 'matplotlib' is included in the Pyodide distribution, but it is not installed.
You can install it by calling:
  await micropip.install("matplotlib") in Python, or
  await pyodide.loadPackage("matplotlib") in JavaScript
See https://pyodide.org/en/stable/usage/loading-packages.html for more details.
```

I am not sure how much of a hope there is too fix it in Pyodide. I would agree that this is indeed suprising.

Reply:
Wow, I see. In that case, can we work around these issues by always importing packages that is commonly use in the examples, such as `matplotlib`, `pandas`, etc.?

Reply:
I guess that would be another option indeed, I was trying to add some code only in the notebooks where it was needed but maybe this is not worth the effort ...

Also I opened https://github.com/pyodide/pyodide/issues/3771 to get some feeling whether the underlying issue is fixable.